text/trace/0189-instrumentation-layers-suppression.md

```diff
@@ -0,0 +1,180 @@
+# Instrumentation Layers and Suppression
+
+This document describes approach for tracing instrumentation layers, suppressing duplicate layers and unambiguously enriching spans.
+
+## Motivation
+
+- Provide clarity for instrumentation layers: e.g. DB calls on top of REST API
+- Give mechanism to suppress instrumentation layers for the same convention: e.g. multiple instrumented HTTP clients using each other.
+- Give mechanism to enrich specific spans unambiguously: e.g. HTTP server span with routing information
+
+## Explanation
+
+### Spec changes proposal
+
+- Tracing Semantic Conventions: Each span MUST follow a single (besides general) convention, specific to the call it describes.
+- Trace API: Add `SpanKey` API that

```

Comment:
As discussed today: alternatively consider moving this logic into the SDK. This would involve fewer API changes and probably result in a more robust solution (otherwise we'd need to rely on every library instrumentor properly calling `SpanKey.*.exists`).